### ISSUE TITLE
Some sample code cannot compile due to missing header line

### ISSUE THREAD

**Issue by cd-pkwan** [utelle/wxchartdir#2]
I found that the following sample code cannot compile due to missing a "#include <vector>" line in the code.

trackbox.cpp
tracklegend.cpp
viewportcontroldemo.cpp
zoomscrolltrack.cpp
realtimetrack.cpp

All of the above sample code use the STL vector class, but does not have the "#include <vector>" line. Once I added the "#include <vector>" immediately before the "using namspace std;" line, they compiled and run normally.

**Comment by utelle** [comment]
Of course, I will look into this issue.

Interestingly, I had no compile problems back then when I released wxChartDir. Therefore, it would be very kind of you to provide some additional information: 

- Which compiler on which platform did you use?
- Which version of wxWidgets did you use? Did you compile wxWidgets yourself or did you use pre-built binaries?

Thanks.

**Comment by utelle** [comment]
For the default wxWidgets builds the preprocessor symbol `wxUSE_STD_DEFAULT` - and thus also `wxUSE_STD_CONTAINERS_COMPATIBLY` - is defined with the value `1`. As a consequence the header `<vector>` is automatically included through the header `"wx/wxprec.h"`resp `"wx/wx.h"`. Therefore, for me all samples compile just fine.

Could it be that in your wxWidgets build the above mentioned preprocessor symbols have the value `0`?

Nevertheless, I will adjust the samples to explicitly include the `<vector>` header file.

**Comment by utelle** [comment]
Commit 279eb1ec12630e3a2f76c044ec538ff4280486e3 should fix the issue.

**Comment by cd-pkwan** [comment]
Hi Ulrich,

I am Peter Kwan from Advanced Software Engineering. 

I am using Ubuntu 16.04 (64-bit), and I use the precompiled wxWidgets distributed by Ubuntu, installed with:

sudo apt-get install libwxgtk3.0-dev
sudo apt-get install libwxgtk3.0-0v5-dbg

I also suspect this may have something to do with pre-compiled headers. It may be due to differences among Linux distributions or wxWidget versions.

Regards
Peter Kwan

**Comment by utelle** [comment]
Hi Peter,

> I am Peter Kwan from Advanced Software Engineering.

I suspected that already from the user id `cd-pkwan`. 😃 

I checked the Ubuntu package `libwxgtk3.0-dev`, and indeed it defines `wxUSE_STD_DEFAULT = 0`. So, this explains, why the std header is required for some of the sample applications. As said, the commit 279eb1ec12630e3a2f76c044ec538ff4280486e3 should fix the issue.

Let me know if you detect further issues.

Regards,
Ulrich
